packages/neos-ui-ckeditor-bindings/src/EditorToolbar/LinkIconButton.js

```diff
@@ -159,42 +153,72 @@ class LinkTextField extends PureComponent {
 
     // A node has been selected
     handleValueChange = value => {
-        if (value) {
-            getGuestFrameWindow().NeosCKEditorApi
-                .toggleFormat(this.props.formattingRule, {href: 'node://' + value});
-        } else {
+        if (!value) {
             getGuestFrameWindow().NeosCKEditorApi
                 .toggleFormat(this.props.formattingRule, {href: ''});
         }
+
+        getGuestFrameWindow().NeosCKEditorApi
+        .toggleFormat(this.props.formattingRule, {href: value});
+
+        const options = this.state.searchOptions.reduce((current, option) =>
+            (option.loaderUri === value) ? [Object.assign({}, option)] : current
+        , []);
+
+        this.setState({options, searchOptions: []});
     }
 
     render() {
         return (
             <div className={style.linkIconButton__flyout}>
                 <SelectBox
                     options={this.props.hrefValue ? this.state.options : this.state.searchOptions}
-                    optionValueField="identifier"
-                    value={this.props.hrefValue && stripNodePrefix(this.props.hrefValue)}
+                    optionValueField="loaderUri"
+                    value={this.props.hrefValue}
                     onValueChange={this.handleValueChange}
                     placeholder="Paste a link, or search"
                     displayLoadingIndicator={this.state.isLoading}
                     displaySearchBox={true}
                     setFocus={true}
                     searchTerm={this.state.searchTerm}
                     onSearchTermChange={this.handleSearchTermChange}
-                    optionComponent={WrappedLinkOption}
+                    optionComponent={LinkOption}
                     />
             </div>
         );
     }
 }
 
-class LinkOption extends PureComponent {
+class ImageOption extends PureComponent {
```

Comment:
Just thinking, won't all of these options and most of this logic be needed in the link editor in the inspector? Shouldn't we extract it into some place from where it could be included from both places?